conda_devenv/devenv.py

```diff
@@ -137,7 +137,14 @@ def merge_dependencies_version_specifications(yaml_dict, key_to_merge):
         return
 
     # regex based on https://conda.io/docs/building/pkg-name-conv.html#package-naming-conventions
-    package_pattern = r'^([a-z0-9_\-.]+)\s*(.*)$'
+    package_pattern = (
+        r'^([a-z0-9_\-/.]+::)?'  # optional channel pattern (group 1)
```

Comment:
You can use named groups `(?P<>` instead:

```python
package_pattern = (
    r'^(?P<channel>[a-z0-9_\-/.]+::)?'
    r'(?P<package>[a-z0-9_\-.]+)'
    r'\s*(?P<version>.*)$'
```

and then access as: `m.group('package')`

Reply:
Awesome, much better